devel/githubContrib.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// go run githubContrib.go to update Contributions.txt
+
+// This script goes from org -> repos (skipping forks) -> contributors -> user
+// -> guess/normalize the company and count contribs
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+// Check checks for non nil error and dies upon error
+func Check(err error, msg string) {
+	if err != nil {
+		log.Fatal(msg, err)
+	}
+}
+
+// getToken gets auth token from the env
+func getToken() string {
+	// fmt.Println("in GetToken")
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		log.Fatal("Need to have GITHUB_TOKEN set in the env")
+	}
+	return token
+}
+
+var gAuthToken = getToken() // Any other way to get a static ?
+
+// GitHubAPIURL returns the full v3 rest api for a given path
+func GitHubAPIURL(path string) string {
+	return "https://api.github.com/" + path
+}
+
+// NewGhRequest makes a GitHub request (with Accept and Authorization headers)
+func NewGhRequest(url string) *http.Request {
+	// fmt.Println("in GhRequest")
+	req, err := http.NewRequest("GET", url, nil)
+	Check(err, "Unable to make request")
+	req.Header.Add("Accept", "application/vnd.github.v3+json")
+	req.Header.Add("Authorization", "token "+gAuthToken)
+	req.Header.Add("User-Agent", "githubContribExtractor")
+	return req
+}
+
+// GetBodyForURL gets the body or dies/abort on any error
+func GetBodyForURL(url string) []byte {
+	req := NewGhRequest(url)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	Check(err, "Unable to send request")
+	body, err := ioutil.ReadAll(resp.Body)
+	Check(err, "Unable to read response")
+	succ := resp.StatusCode
+	log.Printf("Got %d : %s for %s", succ, resp.Status, url)
+	if succ != http.StatusOK {
+		os.Exit(1)
+	}
+	return body
+}
+
+// PrettyPrintJSON outputs indented version of the Json body (debug only)
+func PrettyPrintJSON(body []byte) {
+	var out bytes.Buffer
+	err := json.Indent(&out, body, "", "  ")
+	Check(err, "Unable to Indent json")
+	_, err = os.Stdout.Write(out.Bytes())
+	Check(err, "Unable to output json")
```

Comment:
i'd drop the Check() calls here and just return the error. let the caller decide what to do about it:

```
if err := json.Indent(&out, body, "", "  "); err != nil {
   return fmt.Errorf("not able to indent json: %v", err)
}
```